packages/shared/lib/core/network/classes/iscp-chain.class.ts

```diff
@@ -1,13 +1,15 @@
 import Web3 from 'web3'
 
-import { IBlock, IChain, IChainStatus, IIscpChainMetadata } from '../interfaces'
-import { ChainMetadata, Web3Provider } from '../types'
+import { IBlock, IChain, IChainStatus, IIscpChainConfiguration, IIscpChainMetadata } from '../interfaces'
+import { ChainConfiguration, ChainMetadata, Web3Provider } from '../types'
 
 export class IscpChain implements IChain {
     private readonly _provider: Web3Provider
-    private readonly _metadata: IIscpChainMetadata
+    private readonly _configuration: IIscpChainConfiguration
 
-    constructor(payload: IIscpChainMetadata) {
+    private _metadata: IIscpChainMetadata
```

Comment:
is this still needed?

Reply:
My thinking was that we should store it but lazy load then cache it, so the first time that the metadata needs to be retrieved it fetches from the node then it returns the cache on subsequent method calls.